File: poj/1423.cpp
#include<iostream>
#include<algorithm>
#include<cmath>
#include<cstdio>
using namespace std;
const double pi=3.1415926535898;
const double e=exp((double)(1));
int main(){
    int t;
    while(cin>>t){
        while(t--){
            int n;
            cin>>n;
            int ret;
            ret=(int)(log10(sqrt(2.0*pi*n))+n*log10(n/e))+1);
            cout<<ret<<endl;
        }
    }
    return 0;
}

File: poj/1423.java
import java.util.*;
import java.io.*;
import java.math.*;
public class Main{
	public static void main(String[] args){
		Scanner cin=new Scanner(System.in);
		while(cin.hasNext()){
			int t=cin.nextInt();
			for(int cas=1;cas<=t;cas++){
				int n=cin.nextInt();
				int ret;
				ret=(int)(Math.log10(Math.sqrt(2.0*Math.PI*n))+n*Math.log10(n/Math.E))+1;
				System.out.println(ret);
			}
		}
	}
}
